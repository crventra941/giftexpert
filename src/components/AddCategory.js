import { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const handleChange = (e ) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if(inputValue.trim().length > 2) {
      setCategory(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        onChange={handleChange}
        value={inputValue} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}
