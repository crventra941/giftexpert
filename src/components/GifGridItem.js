import PropTypes from 'prop-types';

export const GifGridItem = ({ img }) => {
  const { title, url } = img;

  return (
    <div className="animate__animated animate__fadeIn card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  img: PropTypes.exact({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  })
}
