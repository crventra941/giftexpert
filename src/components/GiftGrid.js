import { GifGridItem } from "./GifGridItem";

import { useFetch } from "../hooks/useFetch"

export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetch(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {
        loading ? <p className="animate__animated animate__flash">Loading  .....</p> : (
          <div className="card-grid">
            {
              images.map((img) => (
                <GifGridItem key={img.id} img={img} />
              ))
            }
          </div>
        )
      }
    </>
  )
}
