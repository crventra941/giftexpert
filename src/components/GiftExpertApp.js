import { useState } from "react";
import { AddCategory } from './AddCategory';
import { GiftGrid } from "./GiftGrid";

export const GiftExpertApp = () => {
  const [categories, setCategories] = useState(['Dragon  Ball']);

  return (
    <div>
      <h2>Gift Expert App</h2>
      <AddCategory setCategory={setCategories} />
      <hr />

      <ol>
        {
          categories.map((category) => (
            <GiftGrid
              key={category}
              category={category} />
            // <li key={category}>{category}</li>
          ))
        }
      </ol>
    </div>
  );
}
