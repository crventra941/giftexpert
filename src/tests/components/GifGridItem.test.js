import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import '@testing-library/jest-dom';

describe('Pruebas en componente <GifGridItem /> ', () => {
  const imgTest = {
    title: "prueba.png",
    url: "Prueba.png"
  }

  const wrapper = shallow(<GifGridItem img={imgTest} />);

  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(imgTest.title);
  });

  test('debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    // console.log(img.props())
    expect(img.prop('src')).toBe(imgTest.url);
    expect(img.prop('alt')).toBe(imgTest.title);

  });

  test('debe tener la clase animate__fadeIn', () => {
    const div = wrapper.find('div');
    // Otra solucion
    /*
     * const className = div.prop('className');
     * expect(className.includes('animate__fadeIn')).toBe(true);
     *
     * */
    const hasClass = div.find('.card').hasClass('animate__fadeIn');
    expect(hasClass).toBe(true);

  });

});

