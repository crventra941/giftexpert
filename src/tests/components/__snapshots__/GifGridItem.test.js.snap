// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGridItem />  debe mostrar el componente correctamente 1`] = `
<div
  className="animate__animated animate__fadeIn card"
>
  <img
    alt="prueba.png"
    src="Prueba.png"
  />
  <p>
    prueba.png
  </p>
</div>
`;
